eng/pipelines/templates/jobs/ci.yml

```diff
@@ -44,17 +26,23 @@ jobs:
     TestPipeline: ${{ parameters.TestPipeline }}
     PipelineArtifactName: ${{ parameters.PipelineArtifactName }}
 
-- ${{ each matrix in parameters.BuildMatrix }}:
-  - ${{ each toolchain in matrix.Toolchains }}:
-    - template: /eng/pipelines/templates/jobs/test.yml
-      parameters:
-        ServiceDirectory: ${{ parameters.ServiceDirectory }}
-        Toolchain: ${{ toolchain }}
-        Matrix: ${{ matrix }}
-        TimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
-
 - template: /eng/pipelines/templates/jobs/analyze.yml
   parameters:
     ServiceDirectory: ${{ parameters.ServiceDirectory }}
-    Toolchain: ${{ parameters.AnalyzeToolchain }}
+    Toolchain: stable
     TimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+
+- template: /eng/common/pipelines/templates/jobs/generate-job-matrix.yml
+  parameters:
+    JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
+    MatrixConfigs: ${{ parameters.MatrixConfigs }}
+    MatrixFilters: ${{ parameters.MatrixFilters }}
+    MatrixReplace: ${{ parameters.MatrixReplace }}
+    CloudConfig:
+      Cloud: public
+    SparseCheckoutPaths:
+      - /*
```

Comment:
```suggestion
```

I don't think you need this? It's just for checking out files that may be necessary for matrix generation outside of eng+service directory.

Reply:
So `ci.yml` will need all the code because any of it may need to be built.  Depending on how cargo handles partial workspaces, we may still need all the cargo.tomls to do the `cargo metadata` call even when working with a single package.

Reply:
To avoid long check-out times, I wonder if we could just check out:
```yaml
- /*
- /sdk/**/Cargo.toml
```